identifyyeastcells.py

```diff
@@ -0,0 +1,1146 @@
+        objects.parent_image = input_image
+
+        if self.__get(F_BACKGROUND, workspace, None) is None and self.background_elimination_strategy == BKG_FIRST:
+            self.__set(F_BACKGROUND, workspace, background_pixels)
+
+        workspace.object_set.add_objects(objects, self.object_name.value)
+
+        # Make outlines
+        outline_image = centrosome.outline.outline(objects.segmented)
+        if self.should_save_outlines.value:
+            out_img = cpi.Image(outline_image.astype(bool),
+                                parent_image = input_image)
+            workspace.image_set.add(self.save_outlines.value, out_img)
+
+        # Save measurements
+
+        workspace.measurements.add_measurement(self.object_name.value, M_OBJECT_FEATURES_OBJECT_QUALITY,
+                                               objects_qualities)
+
+        cpmi.add_object_location_measurements(workspace.measurements,
+                                              self.object_name.value,
+                                              objects.segmented)
+
+        cpmi.add_object_count_measurements(workspace.measurements,
+                                           self.object_name.value, np.max(objects.segmented))
+
+    def set_params_from_ui(self, params):
+        def update_params(next_name, first_name, random_name, ui_value):
+            params["segmentation"]["seeding"]["from"][next_name] = ui_value >= 1 or ui_value <= 0.5
+            params["segmentation"]["seeding"]["from"][first_name] = ui_value > 0.5
+            params["segmentation"]["seeding"]["from"][random_name] = max(0, ui_value - 1)
+
+        params_seeding = params["segmentation"]["seeding"]["from"]
+        params["segmentation"]["steps"] = self.iterations.value
+        update_params("cellBorderRemovingCurrSegments", "cellBorder", "cellBorderRandom", self.seeds_border.value)
+        update_params("cellContentRemovingCurrSegments", "cellContent", "cellContentRandom", self.seeds_content.value)
+        params_seeding["cellBorderRemovingCurrSegmentsRandom"] = params_seeding["cellBorderRandom"]
+        params_seeding["cellContentRemovingCurrSegmentsRandom"] = params_seeding["cellContentRandom"]
+
+        params["segmentation"]["seeding"]["from"]["snakesCentroids"] = self.seeds_centroid.value > 0.0
+        params["segmentation"]["seeding"]["from"]["snakesCentroidsRandom"] = max(0, self.seeds_centroid.value - 1)
+
+        params["segmentation"]["stars"]["points"] = self.contour_points.value
+        params["segmentation"]["stars"]["step"] = 1.0 / self.contour_precision.value
+
+        default_size_weight_average = np.average(params["segmentation"]["stars"]["sizeWeight"])
+        params["segmentation"]["stars"]["sizeWeight"] = list(
+            create_size_weights(default_size_weight_average, self.weights_number.value)
+        )
+
+    def get_ui_params_from_precision(self, ui_precision):
+        def params_to_ui(params_seeding, next_name, first_name, random_name):
```

Comment:
It uses is closely tied with `get_ui_params_from_precision` so I would prefer not to let it loose.